core/src/main/scala/frameless/CatalystSummable.scala

```diff
@@ -0,0 +1,26 @@
+package frameless
+
+/**
+  * When summing Spark doesn't change these types:
+  * - Long       -> Long
+  * - BigDecimal -> BigDecimal
+  * - Double     -> Double
+  *
+  * For other types there are conversions:
+  * - Int        -> Long
+  * - ...
+  */
+trait CatalystSummable[A] {
+  def zero: A
```

Comment:
From what I understand, @kanterov  means that if you try to do a sum on column of an empty dataset you want to get this instead of null that spark uses. Is that correct?